code/modules/hydroponics/spreading/spreading_growth.dm

```diff
@@ -170,6 +147,62 @@
 			chem_regen_cooldown = FALSE
 
 
+/obj/effect/plant/proc/life()
+	var/turf/simulated/T = get_turf(src)
+	if(istype(T))
+		health -= seed.handle_environment(T,T.return_air(),null,1)
+	if(health < max_health)
+		//Plants can grow through closed airlocks, but more slowly, since they have to force metal to make space
+		var/obj/machinery/door/D = (locate(/obj/machinery/door) in loc)
+		if (D)
+			health += rand_between(0,0.5)
+		else
+			health += rand_between(1,2.5)
+		refresh_icon()
+		if(health > max_health)
+			health = max_health
+	else if(health == max_health && !plant && (seed.type != /datum/seed/mushroom/maintshroom))
+		plant = new(T,seed)
+		plant.dir = src.dir
+		plant.transform = src.transform
+		plant.age = seed.get_trait(TRAIT_MATURATION)-1
+		plant.update_icon()
+		if(growth_type==0) //Vines do not become invisible.
+			invisibility = INVISIBILITY_MAXIMUM
+		else
+			plant.layer = layer + 0.1
+
+
+/obj/effect/plant/proc/spread()
+	//spread to 1-3 adjacent turfs depending on yield trait.
+	var/max_spread = between(1, round(seed.get_trait(TRAIT_YIELD)*3/14), 3)
+
+	for(var/i in 1 to max_spread)
+		if(prob(spread_chance))
+			sleep(rand(3,5))
+			if(!neighbors.len)
+				break
+			var/turf/target_turf = pick(neighbors)
+			if(istype(target_turf, /turf/simulated/open)) //openspace handle
+				var/turf/simulated/open/turf_below = GetBelow(target_turf)
+				var/obj/effect/plant/child = new type(get_turf(src),seed,src)
+				after_spread(child, turf_below)
+			else
+				var/obj/effect/plant/child = new type(get_turf(src),seed,src)
+				after_spread(child, target_turf)
+			// Update neighboring squares.
+			for(var/obj/effect/plant/neighbor in range(1,target_turf))
+				neighbor.neighbors -= target_turf
+
+
+//after creation act
+//by default, there goes an animation code
+/obj/effect/plant/proc/after_spread(var/obj/effect/plant/child, var/turf/target_turf)
+	spawn(1) // This should do a little bit of animation.
```

Comment:
timers are generally better unless you're doing a single proc with a bunch of animation steps in which case it'd just be bloat (which would mean sleeps/spawns are better)

Reply:
instead of doing spawn(1) in proc, call the proc using INVOKE_ASYNC()